File: ios/JiuJitsuLessonTracker/JiuJitsuLessonTracker/Support/SampleData.swift
import Foundation
import SwiftData

/// Seed data so the app shows something useful on first launch and in previews.
enum SampleData {

    /// Inserts sample clients with lessons and payments into the context.
    @MainActor
    static func populate(_ context: ModelContext) {
        let cal = Calendar.current
        let today = cal.startOfDay(for: .now)

        func day(_ offset: Int, hour: Int = 17, minute: Int = 0) -> Date {
            let base = cal.date(byAdding: .day, value: offset, to: today) ?? today
            return cal.date(bySettingHour: hour, minute: minute, second: 0, of: base) ?? base
        }

        // MARK: Alex — active blue belt on a 10-pack
        let alex = Client(
            name: "Alex Costa",
            belt: .blue,
            stripes: 2,
            email: "alex.costa@example.com",
            phone: "555-0142",
            startDate: cal.date(byAdding: .month, value: -14, to: today) ?? today,
            notes: "Competing at the next local open. Working takedowns.",
            defaultLessonRate: 80
        )
        context.insert(alex)
        context.insert(Payment(date: day(-20), amount: 800, method: .card, lessonsCovered: 10, note: "10-lesson package", client: alex))
        context.insert(Lesson(date: day(-18), status: .completed, location: "Main mat", techniques: "Single leg, sprawl defense", rate: 80, client: alex))
        context.insert(Lesson(date: day(-11), status: .completed, location: "Main mat", techniques: "Guard retention, hip escape", rate: 80, client: alex))
        context.insert(Lesson(date: day(-4), status: .completed, location: "Main mat", techniques: "Knee cut pass, underhook", rate: 80, client: alex))
        context.insert(Lesson(date: day(2), status: .scheduled, location: "Main mat", techniques: "Leg drag, back takes", rate: 80, client: alex))
        context.insert(Lesson(date: day(9), status: .scheduled, location: "Main mat", rate: 80, client: alex))

        // MARK: Maria — newer white belt, pay-as-you-go
        let maria = Client(
            name: "Maria Santos",
            belt: .white,
            stripes: 3,
            email: "maria.s@example.com",
            phone: "555-0188",
            startDate: cal.date(byAdding: .month, value: -5, to: today) ?? today,
            notes: "Focus on fundamentals and confidence. Prefers morning sessions.",
            defaultLessonRate: 70
        )
        context.insert(maria)
        context.insert(Payment(date: day(-7), amount: 70, method: .cash, lessonsCovered: 1, note: "Single lesson", client: maria))
        context.insert(Lesson(date: day(-7, hour: 9), status: .completed, location: "Studio B", techniques: "Mount escapes, bridge & roll", rate: 70, client: maria))
        context.insert(Lesson(date: day(1, hour: 9), status: .scheduled, location: "Studio B", techniques: "Closed guard basics", rate: 70, client: maria))

        // MARK: Jordan — purple belt prepping for competition
        let jordan = Client(
            name: "Jordan Lee",
            belt: .purple,
            stripes: 0,
            email: "jordan.lee@example.com",
            phone: "555-0119",
            startDate: cal.date(byAdding: .year, value: -4, to: today) ?? today,
            notes: "Advanced. Drilling competition-specific scenarios.",
            defaultLessonRate: 95
        )
        context.insert(jordan)
        context.insert(Payment(date: day(-30), amount: 475, method: .bankTransfer, lessonsCovered: 5, note: "5-lesson package", client: jordan))
        context.insert(Lesson(date: day(-25), status: .completed, location: "Main mat", techniques: "Berimbolo, leg lock entries", rate: 95, client: jordan))
        context.insert(Lesson(date: day(-12), status: .completed, location: "Main mat", techniques: "Pressure passing", rate: 95, client: jordan))
        context.insert(Lesson(date: day(-3), status: .noShow, location: "Main mat", notes: "Did not show — follow up.", rate: 95, client: jordan))
        context.insert(Lesson(date: day(3, hour: 18), status: .scheduled, location: "Main mat", techniques: "Comp simulation rounds", rate: 95, client: jordan))

        try? context.save()
    }

    /// Seeds sample data only if the store has no clients yet.
    @MainActor
    static func seedIfEmpty(_ context: ModelContext) {
        let descriptor = FetchDescriptor<Client>()
        let count = (try? context.fetchCount(descriptor)) ?? 0
        guard count == 0 else { return }
        populate(context)
    }
}

File: app/src/data/sampleData.ts
import { brand } from "../brand";
import { uid } from "../lib/id";
import { DEFAULT_SKILLS } from "./defaultSkills";
import type { Database } from "./DataStore";
import type { Lesson, Payment, Promotion, Student } from "./types";

// Builds a realistic starter dataset so the app is explorable with zero setup.
export function buildSampleData(): Database {
  const today = new Date();
  today.setHours(0, 0, 0, 0);

  const at = (offsetDays: number, hour = 17, minute = 0): string => {
    const d = new Date(today);
    d.setDate(d.getDate() + offsetDays);
    d.setHours(hour, minute, 0, 0);
    return d.toISOString();
  };
  const dateOnly = (offsetDays: number): string => at(offsetDays).slice(0, 10);
  const monthsAgo = (n: number): string => {
    const d = new Date(today);
    d.setMonth(d.getMonth() - n);
    return d.toISOString().slice(0, 10);
  };
  const yearsAgo = (n: number): string => {
    const d = new Date(today);
    d.setFullYear(d.getFullYear() - n);
    return d.toISOString().slice(0, 10);
  };
  const now = new Date().toISOString();

  const alex: Student = {
    id: uid(),
    name: "Alex Costa",
    belt: "Blue",
    stripes: 2,
    email: "alex.costa@example.com",
    phone: "555-0142",
    startDate: monthsAgo(14),
    notes: "Competing at the next local open. Working takedowns.",
    defaultRateCents: 8000,
    isActive: true,
    createdAt: now,
  };
  const maria: Student = {
    id: uid(),
    name: "Maria Santos",
    belt: "White",
    stripes: 3,
    email: "maria.s@example.com",
    phone: "555-0188",
    startDate: monthsAgo(5),
    notes: "Focus on fundamentals and confidence. Prefers morning sessions.",
    defaultRateCents: 7000,
    isActive: true,
    createdAt: now,
  };
  const jordan: Student = {
    id: uid(),
    name: "Jordan Lee",
    belt: "Purple",
    stripes: 0,
    email: "jordan.lee@example.com",
    phone: "555-0119",
    startDate: yearsAgo(4),
    notes: "Advanced. Drilling competition-specific scenarios.",
    defaultRateCents: 9500,
    isActive: true,
    createdAt: now,
  };

  const L = (
    studentId: string,
    startsAt: string,
    status: Lesson["status"],
    location: string,
    techniques: string,
    rateCents: number,
    notes = "",
  ): Lesson => ({
    id: uid(),
    studentId,
    startsAt,
    durationMinutes: 60,
    status,
    techniques,
    notes,
    location,
    rateCents,
    createdAt: now,
  });

  const P = (
    studentId: string,
    paidOn: string,
    amountCents: number,
    method: Payment["method"],
    lessonsCovered: number,
    note = "",
  ): Payment => ({
    id: uid(),
    studentId,
    paidOn,
    amountCents,
    method,
    lessonsCovered,
    note,
    createdAt: now,
  });

  const Pr = (
    studentId: string,
    promotedOn: string,
    belt: Promotion["belt"],
    stripes: number,
    note = "",
  ): Promotion => ({ id: uid(), studentId, promotedOn, belt, stripes, note });

  const students = [alex, maria, jordan];

  const lessons: Lesson[] = [
    L(alex.id, at(-18), "completed", "Main mat", "Single leg, sprawl defense", 8000),
    L(alex.id, at(-11), "completed", "Main mat", "Guard retention, hip escape", 8000),
    L(alex.id, at(-4), "completed", "Main mat", "Knee cut pass, underhook", 8000),
    L(alex.id, at(2), "scheduled", "Main mat", "Leg drag, back takes", 8000),
    L(alex.id, at(9), "scheduled", "Main mat", "", 8000),
    L(maria.id, at(-7, 9), "completed", "Studio B", "Mount escapes, bridge & roll", 7000),
    L(maria.id, at(1, 9), "scheduled", "Studio B", "Closed guard basics", 7000),
    L(jordan.id, at(-25), "completed", "Main mat", "Berimbolo, leg lock entries", 9500),
    L(jordan.id, at(-12), "completed", "Main mat", "Pressure passing", 9500),
    L(jordan.id, at(-3), "no_show", "Main mat", "", 9500, "Did not show — follow up."),
    L(jordan.id, at(3, 18), "scheduled", "Main mat", "Comp simulation rounds", 9500),
  ];

  const payments: Payment[] = [
    P(alex.id, dateOnly(-20), 80000, "venmo", 10, "10-lesson package"),
    P(maria.id, dateOnly(-7), 7000, "cash", 1, "Single lesson"),
    P(jordan.id, dateOnly(-30), 47500, "zelle", 5, "5-lesson package"),
  ];

  const promotions: Promotion[] = [
    Pr(alex.id, monthsAgo(14), "White", 0, "Started training"),
    Pr(alex.id, monthsAgo(6), "Blue", 0, "Promoted to blue belt"),
    Pr(alex.id, monthsAgo(2), "Blue", 2, "Second stripe"),
    Pr(maria.id, monthsAgo(5), "White", 0, "Started training"),
    Pr(maria.id, monthsAgo(1), "White", 3, "Third stripe"),
    Pr(jordan.id, yearsAgo(1), "Purple", 0, "Promoted to purple belt"),
  ];

  const skills = DEFAULT_SKILLS.map((s) => ({ id: uid(), ...s }));

  const studentSkills = [
    { id: uid(), studentId: maria.id, skillId: skills[0].id, status: "proficient" as const, updatedAt: now },
    { id: uid(), studentId: maria.id, skillId: skills[1].id, status: "drilling" as const, updatedAt: now },
    { id: uid(), studentId: alex.id, skillId: skills[4].id, status: "drilling" as const, updatedAt: now },
    { id: uid(), studentId: alex.id, skillId: skills[0].id, status: "proficient" as const, updatedAt: now },
  ];

  return {
    profile: {
      id: "local-owner",
      businessName: brand.defaultBusinessName,
      defaultRateCents: 8000,
      currency: "USD",
      timezone: Intl.DateTimeFormat().resolvedOptions().timeZone || "UTC",
    },
    students,
    lessons,
    payments,
    promotions,
    skills,
    studentSkills,
  };
}
